client/main.lua

```diff
@@ -615,26 +658,41 @@ RegisterNetEvent('qbx_vehicleshop:client:swapVehicle', function(data)
     local dataTargetVehicle = sharedConfig.shops[shopName].showroomVehicles[data.targetVehicle]
     if dataTargetVehicle.vehicle == data.toVehicle then return end
 
-    local closestVehicle = lib.getClosestVehicle(dataTargetVehicle.coords.xyz, 5, false)
-    if not closestVehicle then return end
-
     if not IsModelInCdimage(data.toVehicle) then
         lib.print.error(('Failed to find model for "%s". Vehicle might not be streamed?'):format(data.toVehicle))
         return
     end
 
-    DeleteEntity(closestVehicle)
-    while DoesEntityExist(closestVehicle) do
-        Wait(50)
+    local points = lib.points.getAllPoints()
+    local vehPoint = nil
+    for i = 1, #points do
+        local point = points[i]
+        if point.coords == dataTargetVehicle.coords.xyz then
+            vehPoint = point
+            break
+        end
     end
 
-    local veh = createShowroomVehicle(data.toVehicle, dataTargetVehicle.coords)
-
-    dataTargetVehicle.vehicle = data.toVehicle
+    if not vehPoint then return end
 
-    if config.useTarget then
-        createVehicleTarget(shopName, veh, data.targetVehicle)
+    if vehPoint.veh then
+        if DoesEntityExist(vehPoint.veh) then
+            DeleteEntity(vehPoint.veh)
+        end
+        if config.useTarget then
+            exports.ox_target:removeLocalEntity(vehPoint.veh, 'vehicleshop:showVehicleOptions')
+        else
+            vehPoint.boxZone:remove()
+        end
     end
+    vehPoint:remove()
+
+    createShowroomVehiclePoint({
+        coords = dataTargetVehicle.coords,
+        shopName = shopName,
+        vehiclePos = data.targetVehicle,
+        model = data.toVehicle
+    })
```

Comment:
```suggestion
    vehPoint:onExit()
    dataTargetVehicle.vehicle = data.toVehicle
    vehPoint.model = data.toVehicle
    if vehPoint.currentDistance <= vehPoint.distance then
        vehPoint:onEnter()
    end
```
Can be simplified to this.
- `onExit` will remove the zone/target and vehicle entity (and some additional cleanup).
- Changing `dataTargetVehicle.vehicle` will make test drives start for the new model instead of the one in the config.
- Changing `vehPoint.model` will use that model next time `onEnter` is ran.
- `onEnter` will create the vehicle entity and the zone/target (only if the player is already inside the point).